### DISCUSSION TITLE
[v5] node-fetch in ace command

### DISCUSSION THREAD

**Discussion by melokki** [adonisjs/core#3560] [Help] [answered]
Hello community

I am trying to create a new ace command and I am having some issues with a import. I know that in commands the top imports are not supported, but I still want to import a package into my command.
I am trying to import the `node-fetch` module
I tried something like this:

```ts
const { default: fetch } = await import('node-fetch')
```
but it does not work

```
  Error 

 require() of ES Module /node_modules/node-fetch/src/index.js from commands/NotifyDocumentExpire.ts not supported.
Instead change the require of index.js in commands/NotifyDocumentExpire.ts to a dynamic import() which is available in all CommonJS modules.
```

Any suggestions?

**Comment by Julien-R44** [comment, marked as answer]
Hello,
the latest versions of node-fetch are ESM-only, as specified in the README: 
https://github.com/node-fetch/node-fetch#commonjs

So downgrade to node-fetch@2 and it should be fine.
Btw, you can totally have an import for node-fetch on top level. Only IoC dependent imports are not possible on top-level.

**Reply by melokki** [reply]
Yes, it appears to work with `node-fetch@2`
Thank you very much :)
